### ISSUE TITLE
Postgres relation query failing because of relation to non primary key

### ISSUE THREAD

**Issue by from-nibly** [nocodb/nocodb#729]
**Please enter the following details**

Node: **v12.22.1**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **pg**
PackageVersion: **0.11.11**

**Steps To Reproduce**
I have two tables that relate to each other in postgres.

Table 1 has a primary key `ID` and another bit of information: `key`,
Table 2 has a primary key of `key`
There is a relation between these two tables on `key` 

When I try to view table 1 it fails because it's adding a where statement to check if `key = <value of an id>` it's throwing because these two columns are different types (key=bigint, id=uuid).

last bit of the error
```
 where "key" = $49 limit $50) - invalid input syntax for type bigint: "4c8277b5-0742-4ad4-932e-001636959608"
```

**Expected behavior**
To join on the join columns and not on primary keys.

**Comment by DOFfactory** [comment]
Total noob here, so apologies if I am posting it under the incorrect heading, nevertheless I wonder whether my issue is related to yours. Also running postrgesql as database. I have the following table:

![image](https://user-images.githubusercontent.com/10846169/143503117-f59fea90-5bae-4e8f-bfc3-51e748bd6d3c.png)

there is an SHARED_ID, a NAME, and a corresponding VALUE. I set SHARED_ID as a primary value. I was wondering whether it would not be more meaningful to "populate" the rows based on SHARED_ID, where the heading can be the NAME with below the corresponding list of VALUES (respective to SHARED_IDs).

Basically I want to achieve something similar than what you have on the screenshot in your front page here on github...

**Comment by o1lab** [comment]
@DOFfactory Its not clear from the description, our discord would be a good place to discuss these. I guess, you may want to set NAME as primary value. That should do it.

**Comment by DOFfactory** [comment]
Basically, what I want is:

```
               | NAME = citationkey | NAME = year  | ...
SHARED_ID = #1 | citationkey_VALUE  | year_VALUE   | ...
SHARED_ID = #2 | citationkey_VALUE  | year_VALUE   | ...
...
```
instead of
```
SHARED_ID | NAME        | VALUE
#1        | citationkey | value
#1        | year        | value
...
...
#2        | citationkey | value
#2        | year        | value
...
```

**Comment by dstala** [comment]
@DOFfactory - please reopen issue or create a discussion thread in discord. closing this, as its related to very old release.
